src/handle/root.rs

```diff
@@ -32,7 +32,7 @@ unsafe impl Sync for NapiRef {}
 /// A `Root<T>` may be sent across threads, but the referenced object may
 /// only be accessed on the JavaScript thread that created it.
 pub struct Root<T> {
-    internal: NapiRef,
+    internal: Option<NapiRef>,
```

Comment:
It took me a while to understand the use of `Option` and convince myself the `unwrap()` is safe. It might help to just document the `Option` is only used to track whether the root has been explicitly dropped via `drop(cx)` /`into_inner()`.

Reply:
That's a good idea. I kept trying to find a good way to implement this, but no matter what I did, it was complicated. `ManuallyDrop` require `unsafe` for _partial_ dropping, which is what we need.

Do you think it would be better with a `bool`? That would remove the `unwrap`.

Reply:
No, I think the `unwrap` is fine with a comment.